From: Liam Howlett <liam.howlett@oracle.com>
Subject: mmap: remove unroll from vma_expand()
Date: Tue, 19 Jul 2022 18:50:01 +0000

Undoing the anon vma work is no necessary since it will be handled by
the caller in the freeing of the VMA, or in the OOM that is happening.

Link: https://lkml.kernel.org/r/20220719184959.3360032-1-Liam.Howlett@oracle.com
Fixes: d8bb385648a5 (mm/mmap: use advanced maple tree API for mmap_region())
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Suggested-by: Hugh Dickins <hughd@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mmap.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/mm/mmap.c~mm-mmap-use-advanced-maple-tree-api-for-mmap_region-fix
+++ a/mm/mmap.c
@@ -623,7 +623,6 @@ inline int vma_expand(struct ma_state *m
 	struct anon_vma *anon_vma = vma->anon_vma;
 	struct file *file = vma->vm_file;
 	bool remove_next = false;
-	bool anon_cloned = false;
 
 	if (next && (vma != next) && (end == next->vm_end)) {
 		remove_next = true;
@@ -635,7 +634,6 @@ inline int vma_expand(struct ma_state *m
 			error = anon_vma_clone(vma, next);
 			if (error)
 				return error;
-			anon_cloned = true;
 		}
 	}
 
@@ -718,8 +716,6 @@ inline int vma_expand(struct ma_state *m
 	return 0;
 
 nomem:
-	if (anon_cloned)
-		unlink_anon_vmas(vma);
 	return -ENOMEM;
 }
 
_
